akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -259,14 +256,17 @@ import scala.util.control.NonFatal
         case est @ Establishing(_, false, OptionVal.None) ⇒
           if (VirtualProcessor.Debug) println(s"VirtualPublisher#$hashCode($est).onError(${t.getMessage}), loop")
           if (!compareAndSet(est, est.copy(onErrorBuffered = OptionVal.Some(ex)))) rec(ex)
```

Comment:
This is where the actual bug being fixed was - if establishing, and the error was successfully buffered, then an NPE wasn't thrown if error was null.